src/verifast1.ml

```diff
@@ -3280,15 +3701,38 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
         in
         let ms = List.map (fun (sign, (tn', lm, gh, rt, xmap, pre, post, epost, terminates, fb', v, abstract, mtparams)) ->
           (* Replace the type parameters with their concrete type*)
-          let methodTargEnv = match zip mtparams targs with Some(tenv) -> tenv | None -> flatmap (fun t -> [(t,javaLangObject)]) mtparams in
+          let methodTargEnv = match zip mtparams targs with
+            Some(tenv) -> tenv
+          | None ->
+            if targs != [] then 
+              List.map (fun tparam -> (tparam, RealTypeParam tparam)) mtparams

```

Comment:
Fout (was al fout in vorige PR): als er type-argumenten opgegeven worden, en die komen niet overeen met het aantal typeparameters, dan is deze methode geen kandidaat voor resolutie.

Reply:
Added a filter that filters the methods. If any type arguments are provided there should be as many as there are type parameters. If none are provided, inference is applied.